### DISCUSSION TITLE
Version 4.9

### DISCUSSION THREAD

**Discussion by joakimeriksson** [contiki-ng/contiki-ng#2531] [General]
# Press release
We are releasing version 4.9 of Contiki-NG. This release adds a new platform (gecko / efr32), and support
for link-time-optimization that can reduce the binary size up to 50% on some examples. Several security related fixes both for software security (e.g. more tests for argument issues) and protocol/communication security (avoiding buffer overruns, etc).

Various parts of the user experience for developers have been improved in this release:
* Docker image updated to Ubuntu 22.04.

## Find out more at:

* GitHub repository: https://github.com/contiki-ng/contiki-ng
* Documentation: https://contiki-ng.readthedocs.io/en/release-v4.9
* Web site: http://contiki-ng.org
* Nightly testbed runs: https://contiki-ng.github.io/testbed

## Engage with the community:

* Contiki-NG tag on Stack Overflow: https://stackoverflow.com/questions/tagged/contiki-ng
* Github Discussions: https://github.com/contiki-ng/contiki-ng/discussions
* Gitter: https://gitter.im/contiki-ng
* Twitter: https://twitter.com/contiki_ng

The Contiki-NG team

### Update from JDOM 1 to JDOM 2

JDOM was upgraded from version 1 to version 2 ([#784](https://github.com/contiki-ng/cooja/pull/784)).
This requires some source code updates, but since Cooja uses such a small subset
of the JDOM API, the update can be done automatically with the command:

```bash
find <directory> -name \*.java -exec perl -pi -e 's#import org.jdom.#import org.jdom2.#g' {} \;
```

### Avoid starting the AWT thread in headless mode

Cooja will no longer start plugins that extend `VisPlugin` in headless mode
to avoid starting the AWT thread. Plugins that should run in both GUI mode
and headless mode need to be updated to keep the JInternalFrame internal.
Examples for PowerTracker and other plugins can be found in the PR
([#261](https://github.com/contiki-ng/cooja/pull/261)).

### Contiki-NG

* Support for link-time-optimization in the build system ([#2077](https://github.com/contiki-ng/contiki-ng/pull/2077))
* Consolidate configuration defines of TI platforms. Prefixes `CCXXWARE_CONF` and `CC26XX_UART_CONF` are deprecated in favor of `CCFG_CONF` and `TI_UART_CONF`. ([#2311](https://github.com/contiki-ng/contiki-ng/pull/2311), [#2387](https://github.com/contiki-ng/contiki-ng/pull/2387))
* Optimize channel offsets in Orchestra to reduce collisions. Note! This changes the default channel offsets in Orchestra. Deployments with a mix of nodes with default configuration at v4.9 and earlier releases may not work, or see poor performance. Adjustments to `ORCHESTRA_*_CHANNEL_OFFSET` may be needed. ([#1673](https://github.com/contiki-ng/contiki-ng/pull/1673))
* API change TSCH: renamed get and remove tsch link functions: (timeslot -> offset in some APIs).
* New platform: gecko or EFR32 - with two supported boards.

All [commits](https://github.com/contiki-ng/contiki-ng/compare/release/v4.8...develop) since v4.8.

### Cooja

* Mobility plugin added to Cooja ([#768](https://github.com/contiki-ng/cooja/pull/768))
* Replace log4j with SLF4J+Logback
* Use foreign function instead of JNI
* Use Gradle for builds
* Java 17

All [commits](https://github.com/contiki-ng/cooja/compare/630e719d01d3...master) since v4.8.

<hr /><em>This discussion was created from the release <a href='https://github.com/contiki-ng/contiki-ng/releases/tag/release/4.9'>Version 4.9</a>.</em>

**Comment by rahav** [comment]
Dear researchers and open-source contributors,

I am writing this note with immense appreciation for your exceptional dedication and commitment to the project's development. Your collective brain trust and countless hours of hard work have culminated in the release of the new version, and I cannot express enough how thankful I am. After trying this new release for a few days, it is stable and allows the rest of us to do our own research.

THANK YOU
